osu.Game/Database/BackgroundDataStoreProcessor.cs

```diff
@@ -86,6 +92,69 @@ protected override void LoadComplete()
             });
         }
 
+        /// <summary>
+        /// Go through every beatmap and calculate the audio normalization values if they are missing.
+        /// </summary>
+        private void processAudioNormalization()
+        {
+            int beatmapsCount = 0;
+            int beatmapsToProcess = 0;
+
+            realmAccess.Run(r =>
+            {
+                foreach (BeatmapSetInfo beatmapSetInfo in r.All<BeatmapSetInfo>().Where(b => !b.DeletePending))
+                {
+                    beatmapsCount += beatmapSetInfo.Beatmaps.Count;
+                    beatmapsToProcess += beatmapSetInfo.Beatmaps.Count(b => b.AudioNormalization == null);
+                }
+            });
+
+            Logger.Log("beatmaps to process: " + beatmapsToProcess);
+            if (beatmapsToProcess == 0)
+            {
+                Logger.Log("No audio normalization processing required.");
+                return;
+            }
+
+            var notification = showProgressNotification(beatmapsCount, "Verifying loudness level for beatmaps", "all loudness levels have been verified");
+            int processedCount = 0;
+
+            realmAccess.Run(r =>
+            {
+                r.Refresh();
+
+                foreach (BeatmapSetInfo beatmapSetInfo in r.All<BeatmapSetInfo>().Where(b => !b.DeletePending))
+                {
+                    foreach (BeatmapInfo beatmapInfo in beatmapSetInfo.Beatmaps)
+                    {
+                        if (notification?.State == ProgressNotificationState.Cancelled)
+                            break;
+
+                        updateNotificationProgress(notification, processedCount, beatmapsCount);
+                        processedCount++;
+
+                        sleepIfRequired();
+                        if (beatmapInfo.AudioNormalization != null) continue;
+
+                        realmAccess.Write(realm =>
+                        {
+                            AudioNormalization audioNormalization = new AudioNormalization(beatmapInfo, beatmapSetInfo, new RealmFileStore(realmAccess, storage!));
+                            beatmapInfo.AudioNormalization = audioNormalization;
+                            realm.Add(audioNormalization.PopulateSet(beatmapInfo, beatmapSetInfo), true);
+                        });
+                        beatmapManager.GetWorkingBeatmap(beatmapInfo, true);
```

Comment:
You're looking for the `Invalidate()` that `BeatmapUpdater` does.

Reply:
I spent a good 2 hours yesterday trying to figure out how I can get Invalidate from `BeatmapUpdater` into `BackgroundDataStoreProcessor`. If I wanted to make invalidate public, that would require editing the interface. I don't feel that changing things to public without understanding what that might bring is wise, so I decided to use this as a workaround to run Invalidate without having to mess with access modifiers.

If it would be better and wouldn't cause issues, I can make `Invalidate()` public and call it here instead.